packages/mui-system/src/Stack/createStack.tsx

```diff
@@ -135,8 +135,12 @@ export const style = ({ ownerState, theme }: StyleFunctionProps) => {
         return { gap: getValue(transformer, propValue) };
       }
       return {
-        '& > :not(style) ~ :not(style)': {
+        // The useFlexGap={false} implement relies on each child to give up control of the margin.
+        // We need to reset the margin to avoid double spacing.
+        '& > :not(style):not(style)': {
           margin: 0,
```

Comment:
We could also consider the path to not reset the margin in the first place. It might be a bit more breaking though that this change but this makes more sense to get close to `useFlexGap`.